lib/matplotlib/axis.py

```diff
@@ -1823,9 +1823,12 @@ def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
         Parameters
         ----------
         ticks : list of floats
-            List of tick locations.
+            List of tick locations.  The axis `.Locator` is replaced by a
+            `.FixedLocator`.  Some default tick formatters will not label
+            arbitrary ticks, so pass *label* to ensure labels are visible.
```

Comment:
I looked a bit into my two bullet-point suggestions and it's not trivial. So I withdraw that proposal.

I think this is the best compromize:

```suggestion
            `.FixedLocator`.
            
            Some tick formatters will not label arbitrary tick positions;
            e.g. log formatters only label decade ticks by default. In
            such a case you can set a formatter explicitly on the axis
            using `.Axis.set_major_formatter` or provide formatted
            *labels* yourself.
```

Reply:
Yes, I think this is a very good, clear text.